src/yaml.cr

```diff
@@ -147,4 +148,28 @@ module YAML
   def self.dump(object, io : IO)
     object.to_yaml(io)
   end
+
+  # Determines if the scaler is reserved
+  def self.reserved_scalar(value : String)
+    case value
```

Comment:
Ah, OK. I just checked the spec and you are right. Ruby seems to implement it wrong:

```
require "yaml"

YAML.load("[nUll, yEs, 1, 2]") # => [nil, true, 1, 2]
```